torchgeo/datasets/etci2021.py

```diff
@@ -0,0 +1,249 @@
+        """
+        return len(self.files)
+
+    def _load_files(self, root: str, split: str) -> List[Dict[str, str]]:
+        """Return the paths of the files in the dataset.
+
+        Args:
+            root: root dir of dataset
+            split: subset of dataset, one of [train, val, test]
+
+        Returns:
+            list of dicts containing paths for each pair of vv, vh,
+            water body mask, flood mask (train/val only)
+        """
+        files = []
+        directory = self.metadata[split]["directory"]
+        folders = sorted(glob.glob(os.path.join(root, directory, "*")))
+        folders = [os.path.join(folder, "tiles") for folder in folders]
+        for folder in folders:
+            vvs = glob.glob(os.path.join(folder, "vv", "*.png"))
+            vhs = glob.glob(os.path.join(folder, "vh", "*.png"))
+            water_masks = glob.glob(os.path.join(folder, "water_body_label", "*.png"))
+
+            if split == "test":
+                flood_masks = [""] * len(water_masks)
+            else:
+                flood_masks = glob.glob(os.path.join(folder, "flood_label", "*.png"))
+
+            for vv, vh, flood_mask, water_mask in zip(
+                vvs, vhs, flood_masks, water_masks
+            ):
+                files.append(
+                    dict(vv=vv, vh=vh, flood_mask=flood_mask, water_mask=water_mask)
+                )
+        return files
+
+    def _load_image(self, path: str) -> Tensor:
+        """Load a single image.
+
+        Args:
+            path: path to the image
+
+        Returns:
+            the image
+        """
+        filename = os.path.join(path)
+        with Image.open(filename) as img:
+            array = np.array(img.convert("RGB"))
+            tensor: Tensor = torch.from_numpy(array)  # type: ignore[attr-defined]
+            # Convert from HxWxC to CxHxW
+            tensor = tensor.permute((2, 0, 1))
+            return tensor
+
+    def _load_target(self, path: str) -> Tensor:
+        """Load the target mask for a single image.
+
+        Args:
+            path: path to the image
+
+        Returns:
+            the target mask
+        """
+        filename = os.path.join(path)
+        with Image.open(filename) as img:
+            array = np.array(img.convert("L"))
+            tensor: Tensor = torch.from_numpy(array)  # type: ignore[attr-defined]
+            tensor = torch.clip(tensor, min=0, max=1)  # type: ignore[attr-defined]
+            tensor = tensor.to(torch.long)  # type: ignore[attr-defined]
+            return tensor
+
+    def _check_integrity(self) -> bool:
+        """Checks the integrity of the dataset structure.
+
+        Returns:
+            True if the dataset directories and split files are found, else False
+        """
+        directory = self.metadata[self.split]["directory"]
+        dirpath = os.path.join(self.root, directory)
+        if not os.path.exists(dirpath):
+            return False
+        return True
```

Comment:
It might be better to check the md5sum of the downloaded files instead of just checking whether a directory exists. There is an open issue on the best way to handle this, not sure what that is just yet though. We'll prob refactor this later, so no need to change it for now.

Reply:
There are many images so wouldn't this create a large list of md5s we have to check? I agree with you though. We can discuss it further in the issue.

Reply:
You could just validate the zip file itself. See #99 for my thoughts on an "ideal" behavior.